src/ResourceStore.php

```diff
@@ -227,15 +229,24 @@ public function peek($type, $id)
 
     public function save(Resource $resource)
     {
-        $method = ($resource->getId() == '') ? 'POST' : 'PATCH';
+        $method = 'POST';
+        $json = $resource->encode();
```

Comment:
We are throwing away this potentially expensive serialization if the resource has an id on the next line. This should be done with an if/else block.